miasm2/core/graph.py

```diff
@@ -229,3 +231,45 @@ def compute_postdominators(self, leaf):
                                                 self.reachable_parents,
                                                 self.successors_iter,
                                                 self.predecessors_iter)
+
+    def compute_immediate_dominators(self, head):
+        """
+        Compute the immediate dominators of the graph
+
+        Depends on the implementation characteristics of reachable_parents.
+        Modifications could break this algorithm.
+        """
+        dominators = self.compute_dominators(head)
+        idoms = {}
+
+        for node in dominators:
+            for predecessor in self.reachable_parents(node):
```

Comment:
Just an idea:

If  ```walk_dominators``` would be a function of DiGraph (mainly replacing the paratemeter ```graph``` with ```self```), then that line could be replaced with
```python
for predecessor in self.walk_dominators(node, dominators):
```
In that case, [the second property](https://github.com/cea-sec/miasm/pull/160#issuecomment-103938202)  would be fulfilled and the dependence on ```reachable_parents``` would be gone.